search-commons/src/main/java/no/unit/nva/search2/common/builder/OpensearchQueryRange.java

```diff
@@ -36,14 +44,58 @@ protected Stream<Entry<K, QueryBuilder>> queryAsEntryStream(K key, String... val
                 .queryName("LessThan-" + key.asCamelCase());
             case BETWEEN -> QueryBuilders
                 .rangeQuery(searchField)
-                .from(firstParam)
-                .to(secondParam)
+                .from(firstParam, true)
+                .to(secondParam, true)
                 .queryName("Between-" + key.asCamelCase());
             default -> throw new IllegalArgumentException(OPERATOR_NOT_SUPPORTED);
         });
     }
 
-    private String valueOrNull(String value) {
-        return value.isBlank() ? null : value;
+    private String getSecondParam(String[] values, K key) {
+        return values.length == 1 ? null :
+            valueOrNull(values[1])
+                .map(date -> expandDateLast(date, key))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private String getFirstParam(String[] values, K key) {
+        return
+            valueOrNull(values[0])
+                .map(date -> expandDateFirst(date, key))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private Stream<String> valueOrNull(String value) {
+        return Stream.ofNullable(value.isBlank() ? null : value);
+    }
+
+    private String expandDateFirst(String date, K key) {
+        var retval = date;
+        if (key.fieldType() != ParameterKind.DATE) {
+            return retval;
+        }
+        if (retval.length() == 4) {
+            retval += "-01";
+        }
+        if (retval.length() == 7) {
+            retval += "-01";
+        }
+        return retval;
```

Comment:
Would it be necessary to validate input length as well? I.e. should `retval == "20240503"` be valid? Not sure how things work, or if the docs specify the specific format, but maybe worth a thought.

Reply:
All parameters are validated before their use (in ParameterValidator.java)